os/fs/smartfs/smartfs_utils.c

```diff
@@ -1579,84 +1579,349 @@ int smartfs_countdirentries(struct smartfs_mountpt_s *fs, struct smartfs_entry_s
 }
 
 /****************************************************************************
- * Name: smartfs_truncatefile
+ * Name: smartfs_sync_internal
  *
- * Description: Truncates an existing file on the device so that it occupies
- *              zero bytes and can be completely re-written.
+ * Description: 
+ *   Synchronize the file state on disk to match internal, in-memory state.
  *
  ****************************************************************************/
 
-int smartfs_truncatefile(struct smartfs_mountpt_s *fs, struct smartfs_entry_s *entry, FAR struct smartfs_ofile_s *sf)
+int smartfs_sync_internal(FAR struct smartfs_mountpt_s *fs, FAR struct smartfs_ofile_s *sf)
```

Comment:
We add journal log in smartfs layer.
When this function was in smartfs layer, it included journal feature, but you removed it...